### ISSUE TITLE
failing to integrate in Ionic project

### ISSUE THREAD

**Issue by jeffleus** [moreaup/cordova-plugin-aws-user-pool#1]
I ran into problems using the JS SDK for Cognito and stumbled across your plugin.  This seems like a good workaround to use native processing to get the tokens.  then, I can use the tokens in the auth header for all calls to my APIG/Lambda/RDS stack on AWS.  But... I am having trouble w/ your plugin.  Do you have some more details on the installation of the plugin on a clean cordova app?

I cloned your repo and then did a symlink plugin add.  Then, I fiddled w/ getting the cordova build for ios to run.  I am currently stuck w/ an ios build from cordova that will not build and deploy in xCode.  Any time to help me trouble shoot?

**Comment by happygts** [comment]
Hi,

The plugin is not yet ready.
I didn't expect someone using it so quickly, there is some of the code only working with my application.
But don't worry, it's working and I will clean in up for global use during the day.
Will tell you when it's done.

**Comment by jeffleus** [comment]
Thanks for the quick reply.  I actually forked and played around w/it today and got some basic stuff working.  I think my use case may be different.  I have a cordova hybrid app already written.  I built the backend on the AWS stack (RDS/Lambda/APIGateway/Cognito).  The problem was the HUGE performance hit to the Cognito login because of big integer processing in UIWebView.  I looked at WKWebView plugin, but couldn't get it to work w/ my project.  So, I turned to the plugin option using the AWS Mobile SDK and stumbled on your project.

For my case, I want to complete a login and then pass the tokens (id/access/refresh) back up to the JS layer and use this in my cordova app.  All of my calls to the APIGateway are protected and require an Authorization header w/ the correct idToken to be used.  I think that your code is using the actually provider paradigm where all the communication (like lambda calls) is happening in the native SDK.  Is this correct?  Or, did I misunderstand your project.

Anyway, for now I just have the native part working.  Next step is to get it running as a plugin for my Ionic/cordova app.  I will share my project when I get a little farther along.  But I have starred/watching your repo and look forward to seeing your updates.  Great work and talk to you soon...
